### ISSUE TITLE
Memory leak in TextEdit

### ISSUE THREAD

**Issue by Sylvain303** [godotengine/godot#37838]
Static memory leak when repeatedly update a `TextEdit.text` value.

**Godot version:**
3.2.1 stable

**OS/device including version:**
Linux xubuntu 20.04

```
Linux lap43 5.4.0-21-generic #25-Ubuntu SMP Sat Mar 28 13:10:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu Focal Fossa (development branch)
Release:	20.04
Codename:	focal
```

**Issue description:**

**Steps to reproduce:**

Add a `TextEdit` and repeatedly update `.text` with some text data.

**Minimal reproduction project:**

https://github.com/Sylvain303/godot-memoryleak-TextEdit

**Comment by Sylvain303** [comment]
Seems related to the undo_history enabled by default or always on:
This seems to fix it for me, because I use it to display text info on the screen. 

https://github.com/Sylvain303/godot-memoryleak-TextEdit/commit/2ad01ce517ce0dcf1fb08c104b8bd028132a92ce

```gdscript
# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta):
	text = big
	clear_undo_history()
```

So maybe not a bug.

**Comment by qarmin** [comment]
History stack should have maximal size e.g. 64, and should not increase memory infinitely.

**Comment by CaptainProton42** [comment]
Stack size seems relatively easy to implement by adding a check like this:

```
if (undo_stack.size() > undo_stack_max_size) {
	undo_stack.pop_front();
}
```

to `TextEdit::_push_current_op()`.
`undo_stack_max_size` could be exposed as a new property in the editor as well.

`TextEdit` also already has a property `undo_enabled` that is not exposed yet, might be an idea to do that as well in case one wants to disable undo completely.

This seems like something I could try myself on if noone else is already doing it.

**Comment by Sylvain303** [comment]
Sorry I'm new to Godot, I used TextEdit as a newline capable textarea, to display live value. And the FPS starts falling quite quickly with 1 core at 100% CPU and memory growing. Because of a lot of changes printed.

I also experienced a text lag in the code editor too, after a whole day of coding, could it be related to the same UI widget, is TextEdit the same for the gdscript code editor?

Yes TextEdit is quite complex component in fact, but for raw output, it's quite easy to setup. I didn't expect to have an undo history. But while looking at the doc in order to fix it me, I found the history guilty feature. :wink:
